integration_test/common/common.go

```diff
@@ -104,30 +107,47 @@ func SliceContains(slice []string, toFind string) bool {
 
 func NewIntegrationMetadataValidator() *validator.Validate {
 	v := validator.New()
-	_ = v.RegisterValidation("onerepresentative", func(fl validator.FieldLevel) bool {
+	_ = v.RegisterValidation("onetrue", func(fl validator.FieldLevel) bool {
+		field := fl.Field()
+		param := fl.Param()
 
-		fieldStr := fl.Field().String()
-		if fieldStr == "" {
-			// Ignore the case where this field is not actually specified or is left empty.
-			return true
+		if param == "" {
+			panic(fmt.Sprintf("onetrue must contain an argument"))
 		}
 
-		metrics, ok := fl.Field().Interface().([]*ExpectedMetric)
-		if !ok {
-			return false
-		}
+		switch field.Kind() {
 
-		if len(metrics) == 0 {
-			return true
-		}
+		case reflect.Slice, reflect.Array:
+			elem := field.Type().Elem()
+
+			if field.Len() == 0 {
+				return true
+			}
 
-		representativeCount := 0
-		for _, m := range metrics {
-			if m.Representative {
-				representativeCount += 1
+			if elem.Kind() == reflect.Ptr {
+				elem = elem.Elem()
 			}
+
+			count := 0
+			for i := 0; i < field.Len(); i++ {
+				elm := reflect.Indirect(field.Index(i))
+
+				f := elm.FieldByName(param)
+
+				if !f.IsValid() {
```

Comment:
I would do this check via the type, rather than on each element of the slice. E.g., something like this:
```Go
		case reflect.Slice, reflect.Array:
			// Ignore the case where this field is not actually specified or is left empty.
			if field.Len() == 0 {
				return true
			}

			elem := field.Type().Elem()
			if elem.Kind() == reflect.Ptr {
				elem = elem.Elem()
			}

			sf, ok := elem.FieldByName(param)
			if !ok {
				panic(fmt.Sprintf("Invalid field name %s", param))
			}
			if sfTyp := sf.Type; sfTyp.Kind() != reflect.Bool {
				panic(fmt.Sprintf("Field %s is %s, not bool", param, sfTyp))
			}

			count := 0
			for i := 0; i < field.Len(); i++ {
				if !reflect.Indirect(field.Index(i)).FieldByName(param).IsZero() {
					count++
				}
			}

			return count == 1
```